qiskit/extensions/standard/boolean_logical_gates.py

```diff
@@ -0,0 +1,131 @@
+        flags = [1 for i in range(len(qr_variables))]
+    else:
+        if len(flags) > len(qr_variables):
+            raise CircuitError('`flags` cannot be longer than `qr_variables`.')
+
+    # check variables
+    # TODO: improve the check
+    if isinstance(qr_variables, (QuantumRegister, list)):
+        variable_qubits = [qb for qb, i in zip(qr_variables, flags) if not i == 0]
+    else:
+        raise ValueError('A QuantumRegister or list of qubits is expected for variables.')
+
+    # check target
+    if isinstance(qb_target, Qubit):
+        target_qubit = qb_target
+    else:
+        raise ValueError('A single qubit is expected for the target.')
+
+    # check ancilla
+    if qr_ancillae is None:
+        ancillary_qubits = []
+    elif isinstance(qr_ancillae, QuantumRegister):
+        ancillary_qubits = list(qr_ancillae)
+    elif isinstance(qr_ancillae, list):
+        ancillary_qubits = qr_ancillae
+    else:
+        raise ValueError('An optional list of qubits or a '
+                         'QuantumRegister is expected for ancillae.')
+
+    all_qubits = variable_qubits + [target_qubit] + ancillary_qubits
+
+    circuit._check_qargs(all_qubits)
+    circuit._check_dups(all_qubits)
+
+    return flags
+
+
+def logical_and(self, qr_variables, qb_target, qr_ancillae, flags=None, mct_mode='no-ancilla'):
+    """
+    Build a collective conjunction (AND) circuit in place using mct.
+
+    Args:
+        self (QuantumCircuit): The QuantumCircuit object to build the conjunction on.
+        qr_variables (QuantumRegister): The QuantumRegister holding the variable qubits.
+        qb_target (Qubit): The target qubit to hold the conjunction result.
+        qr_ancillae (QuantumRegister): The ancillary QuantumRegister for building the mct.
+        flags (list[int]): A list of +1/-1/0 to mark negations or omissions of qubits.
+        mct_mode (str): The mct building mode.
+    """
+    flags = _do_checks(flags, qr_variables, qb_target, qr_ancillae, self)
+    _logical_and(self, qr_variables, flags, qb_target, qr_ancillae, mct_mode)
+
+
+def logical_or(self, qr_variables, qb_target, qr_ancillae, flags=None, mct_mode='basic'):
+    """
+    Build a collective disjunction (OR) circuit in place using mct.
+
+    Args:
+        self (QuantumCircuit): The QuantumCircuit object to build the disjunction on.
+        qr_variables (QuantumRegister): The QuantumRegister holding the variable qubits.
+        flags (list[int]): A list of +1/-1/0 to mark negations or omissions of qubits.
+        qb_target (Qubit): The target qubit to hold the disjunction result.
+        qr_ancillae (QuantumRegister): The ancillary QuantumRegister for building the mct.
+        mct_mode (str): The mct building mode.
+    """
+    flags = _do_checks(flags, qr_variables, qb_target, qr_ancillae, self)
+    _logical_or(self, qr_variables, flags, qb_target, qr_ancillae, mct_mode)
+
+
+QuantumCircuit.AND = logical_and
+QuantumCircuit.OR = logical_or
```

Comment:
all-caps method names are a bit weird. also there is no actual class associated with these gates. 

I think these make sense as part of the circuit library under `qiskit.circuit.library.boolean_logic`. Then if you need to use them as gates anywhere you can just do `big_circuit.append(OR.to_gate())` or something, rather than have these methods.

Reply:
Yes they should go there, I just needed them for the library so I moved them over from Aqua. I'll port these to the logic library but didn't have the time yet 🙂